data/src/main/java/com/microsoft/azure/kusto/data/auth/ApplicationCertificateTokenProvider.java

```diff
@@ -0,0 +1,22 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+package com.microsoft.azure.kusto.data.auth;
+
+import com.microsoft.aad.msal4j.ConfidentialClientApplication;
+import com.microsoft.aad.msal4j.IClientCertificate;
+import org.jetbrains.annotations.NotNull;
+
+import java.net.MalformedURLException;
+import java.net.URISyntaxException;
+
+public class ApplicationCertificateTokenProvider extends ConfidentialAppTokenProviderBase {
+    ApplicationCertificateTokenProvider(@NotNull String applicationClientId, @NotNull IClientCertificate clientCertificate, @NotNull String clusterUrl, String authorityId) throws URISyntaxException {

```

Comment:
Does Java support SNI auth?

Reply:
When I started this, I made a list of what was available in the library and then whittled that list down with Ran and Vladik to decide which ones we wanted. SNI auth either didn't make the first cut (not supported by Java MSAL API), or we didn't want it included.
I'll look up shortly which it was.

Reply:
I don't think it's supported in the MSAL Java API

Reply:
When I originally replaced ADL with MSAL in the Java SDK, the latest Java MSAL API (1.7.0) did not support SNI auth.

However, the very next version (1.7.1) added support for it (see https://github.com/AzureAD/microsoft-authentication-library-for-java/issues/219).

I suppose we should add it - good catch! (But in a separate PR to keep this focused.)
Tagging @vladikbr to confirm